soc/arm/nxp_s32/s32k3/Kconfig.defconfig.series

```diff
@@ -24,6 +24,17 @@ endif
 
 if NET_L2_ETHERNET
 
+# GMAC ethernet driver must be initialized before MDIO driver because
```

Comment:
in fact there's a better way to do this without the need to tweak the init priorities. Let's move `mdio` node one level up, and in the MDIO driver call `Gmac_Ip_EnableMDIO` before each transaction (https://github.com/zephyrproject-rtos/zephyr/blob/00b271c8b5d2e5391cd4714e048d6d8ea2d7db2a/drivers/mdio/mdio_nxp_s32_gmac.c#L123) . Then the MDIO driver can be initialized before Eth driver, because the MDIO transaction parameters are setup just before each transaction.

Moving `mdio` node one level up is also needed when the Eth driver is integrated with the PHY driver, because `emac` node will need to have a reference to the `phy` node, and as it is right now, it will create a circular dependency.

Reply:
Sounds good! Just pushed changes.